Fix ISO code for Canadian French

gunchleoc opened the pull request:
Closes https://github.com/mastodon/mastodon/issues/24538

gunchleoc committed:
Fix ISO code for Canadian French

nschonni commented:
This makes a lot of sense, but also needs to be done on Crowdin, along with the translation file names

vmstan commented:
config/locales/fr-QC.yml needs to be modified as well

gunchleoc commented:
I have opened a discussion on Crowdin to see what the translators for this language say https://crowdin.com/project/mastodon/discussions/311

gunchleoc commented:
I expect this will also need database migrations:

1. Display language setting
2. Default posting language
3. General language filter setting
4. Language filter for follow relationships
5. Post language of individual posts

Release notes also need to inform site admins of the change, in case somebody has set `DEFAULT_LOCALE` to `fr-QC`.

Did I miss anything?

gunchleoc committed:
Update locale configuration

gunchleoc commented:
I have renamed the files and fixed the keys. `fr-QC` was not available in the language filters, so for the code we now only need migrations for:

1. Display language setting
2. Default posting language

Is there some documentation on how to write database migrations?

gunchleoc committed:
Add database migration

gunchleoc commented:
I have found out how to do the migration, so this should now be complete and ready for review. Branch has been rebased.

gunchleoc commented:
Canadian French is not available as a posting language; it is rerouted to French. So, no posts or posting language migrations needed.

What does need doing is an update of the configuration in Crowdin to match the translations pull/push to the new locale code.

mjankowski commented on db/migrate/20230813203012_fix_canadian_french_locale.rb:
```diff
@@ -0,0 +1,25 @@
+  disable_ddl_transaction!
+
+  def up
+    User.where(locale: 'fr-QC').find_in_batches do |users|
```
Could this just be one SQL update, instead of iterating through user objects?

gunchleoc replied:
Done. I don't know how to test the change though, since I've already run the migration. I find the Ruby documentation too hard to navigate on this topic, so I'm basically just frankencoding these from older migrations.

ClearlyClaire replied:
Alright, I can write some migration tests.

ClearlyClaire replied:
Oops, this should be `in_batches` and not `find_in_batches`!

gunchleoc committed:
Use update_all in migration

ClearlyClaire committed:
Add migration tests

ClearlyClaire commented:
FYI, I just pushed a migration test to the PR, please pull if you intend to force-push again!

gunchleoc commented:
> FYI, I just pushed a migration test to the PR, please pull if you intend to force-push again!

Thanks for the test & fix!
